pkg/util/retry/retry.go

```diff
@@ -70,6 +70,11 @@ func StartWithCtx(ctx context.Context, opts Options) Retry {
 	return r
 }
 
+// CurrentAttempt reports the index of the current attempt.
```

Comment:
Can you clarify the message so that it's clear whether it'll return `0` initially or `1`? I also think the word "index" is confusing, but that may be only me.